Add PreferredLanguageCode for customers/getall (#626)
MewsSystems/gitbook-connector-api: operations/customers.md

| `Title` | string [Title](#title) | optional | Title prefix of the customer. |
| `Sex` | string [Sex](#sex) | optional | Sex of the customer. |
| `NationalityCode` | string | optional | ISO 3166-1 code of the [Country](countries.md#country). |

> | `PreferredLanguageCode` | string | optional | Language and culture code of customer's preferred language as set via Operations or API (i.e. without customer's choice). For example: `en-GB`, `fr-CA`. |

Reviewer: I've got to be honest, I don't understand the explanations here. The word 'preferred' suggests that this is what the customer prefers, but that's the opposite of what it says.

> | `PreferredLanguageCode` | string | optional | Language and culture code of customer's preferred language as set via Operations or API (i.e. without customer's choice). For example: `en-GB`, `fr-CA`. |

Reviewer: ```suggestion
| `PreferredLanguageCode` | string | optional | Language and culture code of the customer's preferred language, according to their profile. For example: `en-GB`, `fr-CA`. |
```

> | `LanguageCode` | string | optional | Language and culture code of the customers specified in the booking process, otherwise customers preferred language or native language. E.g. `en-US` or `fr-FR`. |

Reviewer: Likewise here, I don't understand the logic. It didn't seem to matter when there was only one language field, now there are two we need to be clear what the difference is.

Author: The partners were asking for this, because `LanguageCode` field do not rely on 1 thing but has some logic which language it is gonna take as described in the property - first it will look into the customers specified language in the booking process, if that was not provided than it will try to check if there is `preferred language` of the customer and even if that is not available in the system, it will use the native language.

The new field `PreferredLanguageCode` will always take `preferred language` of the customer and nothing else.

Reviewer: Thanks @VBuben for clarifying this. So, my understanding is this:
* `LanguageCode` = "The preferred language of the customer (general). This is the preferred language specified in the booking, else the preferred language of the customer specified in their profile, else the default language for their nationality."
* `PreferredLanguageCode` = "The preferred language of the customer, according to their profile."

In that case, I also have two further questions:
1. When we say "in the booking", given that this field is part of the `Customer` object returned by `Get all customers`, what booking exactly? Is this "the last booking made for this customer", or something else?
2. `LanguageCode` is optional, so if no preferred language is specified in either the booking or the profile, do we actually not include this field, and so the API user should use the default language linked to the nationality, or do we always specify it, in which case why is it optional?

Thanks :-)

Author: @MikeAdamsMews sorry totally forgot about this one.

1. To be honest, I don't know fully how the field is updated. I'm aware of that it can get updated from the reservation customer confirm their reservation. @jnv maybe you have deeper knowing around this? I remember we were looking into it together.
2. Well, it can get null since all fields it takes data from are nullable

> | `LanguageCode` | string | optional | Language and culture code of the customers specified in the booking process, otherwise customers preferred language or native language. E.g. `en-US` or `fr-FR`. |

Reviewer: ```suggestion
| `LanguageCode` | string | optional | Language and culture code of the customer's language, based on multiple sources. These sources include the preferred language specified in internal data based on previous bookings, and the preferred language of the customer specified in their profile. If neither of these sources are present, we use the native language based on the customer's nationality. The format is, for example, `en-US` or `fr-FR`. |
```
